docs/DestroyPlugin.md

```diff
@@ -0,0 +1,73 @@
+## [DestroyPlugin](../src/DestroyPlugin.groovy)
+
+Enable this plugin to use `terraform destroy` to destroy your environment(s).
+
+When this plugin is enabled, the pipeline will follow these steps:  
+1. Run a `terraform plan -destroy` to display which resources will get destroyed.
+2. Ask for human confirmation to proceed with the destroy.
+3. Run the `terraform destroy` command.
+
+
+```
+// Jenkinsfile
+@Library(['terraform-pipeline@v3.10']) _
+
+Jenkinsfile.init(this, env)
+
+// This enables the destroy functionality
+DestroyPlugin.init()
+
+def validate = new TerraformValidateStage()
+
+def destroyQa = new TerraformEnvironmentStage('qa')
+def destroyUat = new TerraformEnvironmentStage('uat')
+def destroyProd = new TerraformEnvironmentStage('prod')
+
+validate.then(destroyQa)
+        .then(destroyUat)
+        .then(destroyProd)
+        .build()
+```
+
+When using this plugin, your pipeline will look something like this:
+
+![DestroyPluginPipeline](../images/destroy-pipeline.png)
+
+---------
+
+## How to destroy environments after deployment
+
+If you wish to run a traditional deployment and then run `terraform destroy`, you can enable the DestroyPlugin after the deployment.
+
+
+```
+def validate = new TerraformValidateStage()
+
+def deployQa = new TerraformEnvironmentStage('qa')
+def deployUat = new TerraformEnvironmentStage('uat')
+def deployProd = new TerraformEnvironmentStage('prod')
+
+// First we deploy our environments
+validate.then(deployQa)
+        .then(deployUat)
+        .then(deployProd)
+        .build()
+
+
+// Now enable the destroy functionality
+DestroyPlugin.init()
+
+def destroyQa = new TerraformEnvironmentStage('qa')
+def destroyUat = new TerraformEnvironmentStage('uat')
+def destroyProd = new TerraformEnvironmentStage('prod')
+
+// Destroy the environments
+validate.then(destroyQa)
+        .then(destroyUat)
+        .then(destroyProd)
+        .build()
+```
+
```

Comment:
I'm not sure I'm a fan of this.  The premise of the library is that once you  call `.build()`, no other code should be executed.  I think it might be a necessary constraint in order to be to effectively create Declarative Pipelines (the syntax is really strict).  Note: [README](https://github.com/manheim/terraform-pipeline#how-to-use) `The design of this library is influenced by the Builder Pattern - your pipeline has been configured, but hasn't been constructed just yet. Finalize and create your pipeline by calling the build() method. This should only be done once - no code should come after calling this method.`


At any rate, I don't think I'm ready to concede on the `.build()` rule just yet.  

Can we  separate this PR into UseCases?  There's 1 UseCase of just terminating all the environments, which is your first example.  Can we do a separate PR to implement the other UseCase of deploying+destroying?  I'd like to talk more about how we  could implement that.